Userland/Libraries/LibWeb/SRI/SRI.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2023, Sam Atkins <atkinssj@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+
+namespace Web::SRI {
+
+// https://w3c.github.io/webappsec-subresource-integrity/#integrity-metadata
+struct Metadata {
+    String algorithm;    // "alg"
+    String base64_value; // "val"
+    String options {};   // "opt"
```

Comment:
Omit `{}`

Reply:
That's there because we don't ever assign to it when creating the Metadata struct, so it needs a default:
```
// 8. Let metadata be the ordered map «["alg" → algorithm, "val" → base64-value]».
//    Note: Since no options are defined (see the §3.1 Integrity metadata), a corresponding entry is not set in metadata.
//    If options are defined in a future version, hash-with-opt-token-list[1] can be utilized as options.
```
I can just assign it to `{}` there instead though if you'd prefer.

Reply:
`String string;` and `String string {};` to the same thing, I'm not following...

Reply:
This is a C++-ism with designated initializers + whatever set of compiler warning we set:
```c++
struct Foo {
    String abc;
    String def;
    String ghi;
};

void bar()
{
    Foo foo {
        .abc = "abc"_short_string,
        .def = "def"_short_string,
    };
}
```

This won't compile because `foo.ghi` isn't initialized, even though it has a default constructor.

Reply:
Thanks, I hate it